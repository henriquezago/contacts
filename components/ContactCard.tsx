import { Contact } from "../pages/api/contacts";

import { Card, Col, Text } from "@nextui-org/react";

type ContactCardProps = {
  contact: Contact;
  onClick: (contact: Contact) => void;
}

export default function ContactCard({ contact, onClick }: ContactCardProps) {
  return (
    <Card isPressable={true} onClick={() => onClick(contact)}>
      <Card.Header css={{ position: "absolute", zIndex: 1, top: 5 }}>
        <Col>
          <Text h4 color="white">
            {contact.name}
          </Text>
        </Col>
      </Card.Header>
      <Card.Divider />
      <Card.Body>
        <Card.Image
          src={contact.avatar}
          objectFit="contain"
          width={240}
          height="100%"
          alt={`${contact.name}'s avatar image`}
        />
      </Card.Body>
    </Card>
  );
}
