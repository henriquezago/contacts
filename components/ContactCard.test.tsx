import { expect } from '@jest/globals';
import { fireEvent, render } from "@testing-library/react";

import ContactCard from './ContactCard';
import { act } from '@testing-library/react';

describe('ContactCard', () => {
  const contact = {
    id: "1",
    avatar: "",
    name: "John Doe",
    email: "",
    phone: "",
    birthday: "",
    createdAt: "",
  };

  test("renders", () => {
    const container = render(
      <ContactCard contact={contact} onClick={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });

  test("should call onClick", () => {
    const onClick = jest.fn();

    const container = render(
      <ContactCard contact={contact} onClick={onClick} />
    );

    act(() => {
      fireEvent.click(container.getByRole('button'));
    })

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});