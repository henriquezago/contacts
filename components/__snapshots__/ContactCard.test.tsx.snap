// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-jxECYO nextui-c-jxECYO-icwRywK-css"
        >
          <div
            class="nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css"
          >
            <h4
              class="nextui-c-PJLV nextui-c-PJLV-ikRtugN-css"
            >
              John Doe
            </h4>
          </div>
        </div>
        <div
          class="nextui-c-bZLhCa nextui-c-bZLhCa-qCmpO-color-default nextui-c-bZLhCa-iiMwZzC-css"
          role="separator"
        />
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ihinyfY-css nextui-image-container nextui-image--ready"
            data-state="ready"
          >
            <img
              alt="John Doe's avatar image"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="ready"
              height="100%"
              src=""
              width="100%"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
      role="button"
      tabindex="0"
    >
      <div
        class="nextui-c-jxECYO nextui-c-jxECYO-icwRywK-css"
      >
        <div
          class="nextui-c-cMTZRp nextui-c-cMTZRp-ijmTDtb-css"
        >
          <h4
            class="nextui-c-PJLV nextui-c-PJLV-ikRtugN-css"
          >
            John Doe
          </h4>
        </div>
      </div>
      <div
        class="nextui-c-bZLhCa nextui-c-bZLhCa-qCmpO-color-default nextui-c-bZLhCa-iiMwZzC-css"
        role="separator"
      />
      <div
        class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
      >
        <div
          class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-ihinyfY-css nextui-image-container nextui-image--ready"
          data-state="ready"
        >
          <img
            alt="John Doe's avatar image"
            class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
            data-state="ready"
            height="100%"
            src=""
            width="100%"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
